src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -42,49 +48,63 @@ export const selectStakingDataByAccountSpecifier = createSelector(
   }
 )
 
-export const selectDelegationCryptoAmountByDenom = createSelector(
+export const selectDelegationCryptoAmountByAssetId = createSelector(
   selectStakingDataByAccountSpecifier,
   selectValidatorAddress,
-  selectDenom,
-  (stakingData, validatorAddress, denom): string | undefined => {
+  selectAssetId,
+  (stakingData, validatorAddress, selectedAssetId): string | undefined => {
```

Comment:
```suggestion
  (stakingData, validatorAddress, selectedAssetId): string => {
```

Reply:
we can also feed selectAllDelegationsCryptoAmountByAssetId this guy in and just index by asset id

Reply:
I did it and renamed to `selectDelegationCryptoAmountByValidatorAndAssetId` because the naming wasn't clear to me, does it looks fine?